app/actions/views/channel.js

```diff
@@ -690,4 +694,70 @@ function loadSidebar(data) {
             dispatch(batchActions(sidebarActions, 'BATCH_LOAD_SIDEBAR'));
         }
     };
+}
+
+export function loadUnreadChannelPosts() {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const unreadChannelIds = getUnreadChannelIds(state);
```

Comment:
build the unread channel Ids from the data received and **exclude** the `currentChannelId`.

The reason for the exclusion is cause the posts for current channel Id is being fetch in `handleSelectChannel` and in `doReconnect` regardless if it has unreads or not.